### PR TITLE
Add FunctionsOfTime base tag

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

Adds a new base tag `domain::Tags::FunctionsOfTime`  used to access the functions of time from the gobal cache. The previous tag `domain::Tags::FunctionsOfTime` is renamed to `domain::Tags::FunctionsOfTimeInitialize` and is derived from the base tag.

The motivation for this is that the tag we use to get the functions of time out of the global cache shouldn't care how the tags were put into the global cache. Currently, there is only one way the functions of time are constructed and put into the cache: using the `domain::Tags::FunctionsOfTimeInitialize` tag (formerly named `domain::Tags::FunctionsOfTime`). This tag is in the FunctionsOfTime library (and hence the Domain library), but only uses domain quantities to construct the functions of time. However, once we add control systems, this will no longer be the case. Functions of time will need to be constructed with control system information as well as domain information. This cannot be done in the `domain::Tags::FunctionsOfTimeInitialize` tag, otherwise the Domain library would depend on the ControlSystem library and create a cyclic dependency (because the ControlSystem library already depends on the Domain). Hence why we needed a base tag for the functions of time. In a future PR, an additional tag derived off the `domain::Tags::FunctionsOfTime` base will be added in the ControlSystem library to handle constructing functions of time with control system information.

A majority of this PR is changing the tag name in tests.

### Upgrade instructions

- When accessing the functions of time from the cache, use the same tag name that has been used before: `domain::Tags::FunctionsOfTime`. (This functionality hasn't changed)
- When adding the functions of time to the cache (or to the DataBox), use the new `domain::Tags::FunctionsOfTimeInitialize` tag name. This could be in a `mutable_global_cache_tags` type alias in actions or components, or in the tags list or `simple_tags` type alias when constructing a DataBox (this is typically used in the testing framework).

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tests/Unit/Domain/FunctionsOfTime/Test_Tags.cpp [hunk 1]
```diff
@@ -18,17 +18,20 @@ struct Metavariables {
 };
 
 SPECTRE_TEST_CASE("Unit.Domain.FunctionsOfTime.Tags", "[Domain][Unit]") {
-  TestHelpers::db::test_simple_tag<Tags::FunctionsOfTime>("FunctionsOfTime");
+  TestHelpers::db::test_base_tag<Tags::FunctionsOfTime>("FunctionsOfTime");
+  TestHelpers::db::test_simple_tag<Tags::FunctionsOfTimeInitialize>(
+      "FunctionsOfTimeInitialize");
```
Comment (nilsvu) [inline tests/Unit/Domain/FunctionsOfTime/Test_Tags.cpp]: I thought the name should be that of the simple tag here

**Review by nilsvu — DISMISSED** [review]

File: tests/Unit/Domain/FunctionsOfTime/Test_Tags.cpp (hunk 1 above)
Comment (knelli2) [reply tests/Unit/Domain/FunctionsOfTime/Test_Tags.cpp]: I'm confused, I don't see an issue. "FunctionsOfTimeInitialize" *is* the name of the simple tag now. "FunctionsOfTime" is the name of the base tag now.

Comment (nilsvu) [reply tests/Unit/Domain/FunctionsOfTime/Test_Tags.cpp]: Oh sorry, what I meant is that I thought both the simple and the base tag would have the same name, presumably that of the base tag. That's how compute tags work. But don't worry about this, it's nothing to be concerned about in this PR.

Comment (nilsdeppe) [reply tests/Unit/Domain/FunctionsOfTime/Test_Tags.cpp]: Compute tags take the name of the simple tags, simple tags tag their own name. I think these simple tags should have a name function that returns `FunctionsOfTime` without the `Initialize` unless the tags are going to be removed after initialization.

Comment (knelli2) [reply tests/Unit/Domain/FunctionsOfTime/Test_Tags.cpp]: No the tags aren't removed after initialization. I added a static name function that returns `FunctionsOfTime` to the simple tag and changed the test accordingly (squashed immediately)

Comment (nilsdeppe) [reply tests/Unit/Domain/FunctionsOfTime/Test_Tags.cpp]: Thanks!

**Review by wthrowe — APPROVED** [review]

**Review by nilsvu — APPROVED** [review]

**Pull request merged**